streaming/src/main/scala/org/apache/spark/streaming/dstream/TransformedDStream.scala

```diff
@@ -38,6 +39,11 @@ class TransformedDStream[U: ClassTag] (
 
   override def compute(validTime: Time): Option[RDD[U]] = {
     val parentRDDs = parents.map(_.getOrCompute(validTime).orNull).toSeq
-    Some(transformFunc(parentRDDs, validTime))
+    val transformedRDD = transformFunc(parentRDDs, validTime)
+    if (transformedRDD == null) {
+      throw new SparkException("Transform function must not return null. " +
+        "Return RDD.empty to return no elements as the result of the transformation.")
```

Comment:
"Return RDD.empty instead to represent no element as the result of transformation" ?